core/src/util/input_output.rs

```diff
@@ -279,6 +280,8 @@ pub struct CheckEmailOutput {
 	pub smtp: Result<SmtpDetails, SmtpError>,
 	/// Details about the email address.
 	pub syntax: SyntaxDetails,
+	// Details about gravatar.
+	pub gravatar: GravatarDetails,
```

Comment:
Instead of creating a top-level struct, let's just simply add a `Option<String>` sub-field under the `misc` field.